tests/components/modbus/test_modbus_binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+"""The tests for the Modbus sensor component."""
+import logging
+
+from homeassistant.components.binary_sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.modbus.const import (
+    CALL_TYPE_COIL,
+    CALL_TYPE_DISCRETE,
+    CONF_ADDRESS,
+    CONF_INPUT_TYPE,
+    CONF_INPUTS,
+)
+from homeassistant.const import CONF_NAME, STATE_OFF, STATE_ON
+
+from .conftest import run_base_test
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def run_sensor_test(hass, use_mock_hub, register_config, value, expected):
+    """Run test for given config."""
+    sensor_name = "modbus_test_binary_sensor"
+    entity_domain = SENSOR_DOMAIN
+    data_array = {
+        CONF_INPUTS: [
+            dict(**{CONF_NAME: sensor_name, CONF_ADDRESS: 1234}, **register_config)
+        ]
+    }
+    await run_base_test(
+        sensor_name,
+        hass,
+        use_mock_hub,
+        data_array,
+        register_config.get(CONF_INPUT_TYPE),
+        entity_domain,
+        value,
+        expected,
+    )
+
+    # Check state
+    entity_id = f"{entity_domain}.{sensor_name}"
+    state = hass.states.get(entity_id).state
+    assert state == expected
+
+
+async def test_coil_true(hass, mock_hub):
```

Comment:
Looks like these tests can use pytest parametrization.

Then we can inline the test helper in the test.

https://docs.pytest.org/en/stable/parametrize.html

Reply:
I like that idea (its new to me), I think a lot of the tests can be handled with this. This PR is merged but I will make an update. Thanks for the idea.